mobile/lib/pages/article_selection_page.dart

```diff
@@ -0,0 +1,142 @@
+    super.key,
+    this.timeout = const Duration(seconds: 15),
+  });
+
+  final Duration timeout;
+
+  static const EdgeInsets _pagePadding = EdgeInsets.symmetric(horizontal: 26);
+
+  // Single URL that returns all articles
+  // TODO: Lazy loading and loading every article not one page of articles
+  static final Uri _infoUrl = Uri.parse('https://radioaktywne.pl/wp-json/wp/v2/posts?_embed=true&page=1&per_page=16');
+
+  Future<List<ArticleInfo>> _fetchArticles() async {
+    final articles = <ArticleInfo>[];
+
+    try {
+      final response = await http.get(_infoUrl).timeout(timeout);
+      final jsonDataList = jsonDecode(response.body) as List<dynamic>;
+
+      for (final jsonData in jsonDataList) {
+        final article = ArticleInfo.fromJson(jsonData as Map<String, dynamic>);
+        articles.add(article);
+      }
+    } catch (_) {
+      articles.add(ArticleInfo.empty());
+    }
+    return articles;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+
+    final shadowColors = <Color>[
+      context.colors.highlightRed,
+      context.colors.highlightYellow,
+      context.colors.highlightGreen,
+      context.colors.highlightBlue,
+    ];
+
+    return RefreshableFetchWidget(
+      onFetch: _fetchArticles,
+      defaultData: [ArticleInfo.empty()],
+      loadingBuilder: (context, snapshot) => const _ArticleSelectionWaiting(),
+      errorBuilder: (context) => const _ArticleSelectionNoData(),
+      builder: (context, articles) {
+        return Padding(
+          padding: const EdgeInsets.only(top: 20, bottom: 20),
+          child: GridView.builder(
+            gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+              crossAxisCount: 2, // This creates two columns
+            ),
+            itemCount: articles.length,
+            itemBuilder: (context, index) {
+              final article = articles[index];
+              return Padding(
+                padding: const EdgeInsets.symmetric(horizontal: 15, vertical: 15),
+                // TODO: Add proper navigation to the article page
+                child: GestureDetector(
+                  onTap: () {
+                    Navigator.push<ArticlePage>(
+                      context,
+                      MaterialPageRoute(builder: (context) => ArticlePage(article: article,)),
+                    );
+                  },
+                  child: ColorShadowedCard2(
+                    shadowColor: shadowColors[index % shadowColors.length],
+                    footer: DefaultTextStyle(
+                      style: context.textStyles.textSmall.copyWith(
+                        color: context.colors.highlightGreen,
+                      ),
+                      child: Padding(
+                        padding: const EdgeInsets.all(5),
+                        child: HtmlWidget(article.title),
+                        ),
+                    ),
+                    child: Image.network(article.thumbnail),
```

Comment:
I modified it to this, however I am not entirely sure that this will work properly.
```Dart
child: Image.network(
  article.thumbnail,
  errorBuilder: (context, error, stackTrace) => Center(
    child: Text(
      context.l10n.imageLoadError,
      style: context.textStyles.textMedium.copyWith(
        color: context.colors.highlightGreen,
      ),
      textAlign: TextAlign.center,
      softWrap: true,
    ),
  ),
),
```

When I change the code inside `article_info.dart` for the thumbnail to get the image that doesn't exist liek so for example (thumbnail appears to be in every article because it is displayed on the website i suppose, however medium size is not present in every article)

```Dart
thumbnail = jsonData['_embedded']['wp:featuredmedia'][0]['media_details']['sizes']['medium']['source_url'] as String,
```

I get this error message:

```
════════ Exception caught by widgets library ═══════════════════════════════════
The method '[]' was called on null.
Receiver: null
Tried calling: []("source_url")
════════════════════════════════════════════════════════════════════════════════
``` 

and the screen looks like this:
<img src="https://github.com/RadioAktywne/ra-app/assets/101600853/ac516060-eb8e-4fed-9e1f-497901b6d26b" alt="Simulator Screenshot - iPhone 15 Pro - 2024-06-16 at 19 20 32" width="300">

which tells me that the error is not caught. I need an advice on this.